incubator/couchdb/templates/statefulset.yaml

```diff
@@ -95,14 +93,10 @@ spec:
             mountPath: /opt/couchdb/etc/default.d
           - name: database-storage
             mountPath: /opt/couchdb/data
-        - name: couchdb-statefulset-assembler
-          image: "{{ .Values.helperImage.repository }}:{{ .Values.helperImage.tag }}"
-          imagePullPolicy: {{ .Values.helperImage.pullPolicy }}
 {{- if .Values.enableSearch }}
         - name: clouseau
           image: "{{ .Values.searchImage.repository }}:{{ .Values.searchImage.tag }}"
           imagePullPolicy: {{ .Values.searchImage.pullPolicy }}
-{{- end }}
 {{- if not .Values.allowAdminParty }}
```

Comment:
This is the proposed change from #15162, so that'll need to be cleaned up into separate commits once that other PR is finalized.